pkg/client/client.go

```diff
@@ -460,22 +461,25 @@ func (c *Client) Fetch(ctx context.Context, request FetchRequest) (res *FetchRes
 			)
 			for {
 				resp, err := stream.Recv()
-				if err == io.EOF {
-					pLog.Info("provider finished fetch", "execution", time.Since(fetchStart).String())
-					for _, fetchError := range partialFetchResults {
-						pLog.Warn("received partial fetch error", parsePartialFetchKV(fetchError)...)
-					}
-					fetchSummaries <- ProviderFetchSummary{
-						ProviderName:          providerConfig.Name,
-						Version:               providerPlugin.Version(),
-						TotalResourcesFetched: totalResources,
-						PartialFetchErrors:    partialFetchResults,
-						FetchErrors:           fetchErrors,
-						FetchResources:        fetchedResources,
-					}
-					return nil
-				}
+
 				if err != nil {
+					st, ok := status.FromError(err)
+					if (ok && st.Code() == gcodes.Canceled) || err == io.EOF {
```

Comment:
I think this should be in a switch case and messages should be printed differently (i.e provider fetch cancelled/finished/failed), would also add to the provider fetch summary its finish state i.e finish/error/cancelled.

Reply:
Couldn't figure out how to use a switch without duplicating a bunch of code.

Reply:
Output now looks like this when we cancel the invocation
```
Provider aws fetch summary: ❌ (canceled) Total Resources fetched: 24    ⚠️ Warnings: 0   ❌ Errors: 0
```